ed_sensor_integration/include/ed_sensor_integration/tools/snapshot.h

```
@@ -0,0 +1,189 @@
+#ifndef ED_SENSOR_INTEGRATION_TOOLS_SNAPSHOT_H_
+#define ED_SENSOR_INTEGRATION_TOOLS_SNAPSHOT_H_
+
+#include <ed/models/model_loader.h>
+#include <ed/world_model.h>
+#include <ed/update_request.h>
+#include <ed/serialization/serialization.h>
+#include <ed/io/json_reader.h>
+
+#include <tue/filesystem/crawler.h>
+
+#include <tue/config/read.h>
+#include <tue/config/reader.h>
+#include <tue/config/data_pointer.h>
+
+#include <rgbd/image.h>
+#include <rgbd/serialization.h>
+
+#include <vector>
+
+namespace ed_sensor_integration {
+
+struct Snapshot
+{
+    rgbd::ImagePtr image;
+    geo::Pose3D sensor_pose;
+};
+
+bool readImage(const std::string& filename, rgbd::ImagePtr& image, geo::Pose3D& sensor_pose)
+{
+    tue::config::DataPointer meta_data;
+
+    try
+    {
+        meta_data = tue::config::fromFile(filename);
+    }
+    catch (tue::config::ParseException& e)
+    {
+        std::cerr << "Could not open '" << filename << "'.\n\n" << e.what() << std::endl;
```

Only use bare prints in tools, not in library.

---

Question: I can see why you would not want this in a library. What would be a better way of handling this. Just ignore the exception or make the print not 'bare' as you call it?